src/dsi.rs

```diff
@@ -0,0 +1,693 @@
+
+        // Polarity
+        dsi.lpcr()
+            .modify(|_, w| w.dep().clear_bit().vsp().clear_bit().hsp().clear_bit());
+
+        // Color coding for the host
+        let lpe = match dsi_config.color_coding_host {
+            ColorCoding::EighteenBitsConfig1 => true,
+            ColorCoding::EighteenBitsConfig2 => true,
+            _ => false,
+        };
+        dsi.lcolcr().modify(|_, w| unsafe {
+            w.lpe()
+                .bit(lpe) // loosely packed: 18bits
+                .colc()
+                .bits(dsi_config.color_coding_host as u8) // 0: 16bit_1, 1: 16bit_2, 2: 16bit_3, 3: 18bit_1, 4: 18bit_2, 5: 24bit
+        });
+
+        // Color coding for the wrapper
+        dsi.wcfgr()
+            .modify(|_, w| unsafe { w.colmux().bits(dsi_config.color_coding_wrapper as u8) });
+
+        dsi.lpmcr().modify(|_, w| unsafe {
+            w.lpsize()
+                .bits(dsi_config.lp_size) // Low power largest packet size
+                .vlpsize()
+                .bits(dsi_config.vlp_size) // Low power VACT largest packet size
+        });
+
+        Ok(DsiHost {
+            dsi,
+            channel: dsi_config.channel,
+            cycles_1ms,
+        })
+    }
+
+    pub fn set_command_mode_transmission_kind(&mut self, kind: DsiCmdModeTransmissionKind) {
+        let is_low_power = match kind {
+            DsiCmdModeTransmissionKind::AllInHighSpeed => false,
+            DsiCmdModeTransmissionKind::AllInLowPower => true,
+        };
+        self.dsi.cmcr().modify(|_, w| {
+            w.gsw0tx()
+                .bit(is_low_power)
+                .gsw1tx()
+                .bit(is_low_power)
+                .gsw2tx()
+                .bit(is_low_power)
+                .gsr0tx()
+                .bit(is_low_power)
+                .gsr1tx()
+                .bit(is_low_power)
+                .gsr2tx()
+                .bit(is_low_power)
+                .glwtx()
+                .bit(is_low_power)
+                .dsw0tx()
+                .bit(is_low_power)
+                .dsw1tx()
+                .bit(is_low_power)
+                .dsr0tx()
+                .bit(is_low_power)
+                .dlwtx()
+                .bit(is_low_power)
+                .mrdps()
+                .bit(is_low_power)
+        });
+        self.dsi.cmcr().modify(|_, w| w.are().clear_bit()); // FIXME: might be incorrect
+    }
+
+    pub fn configure_phy_timers(&mut self, phy_timers: DsiPhyTimers) {
+        let max_time = max(phy_timers.clock_lp2hs, phy_timers.clock_hs2lp);
+        self.dsi
+            .cltcr()
+            .modify(|_, w| unsafe { w.hs2lp_time().bits(max_time).lp2hs_time().bits(max_time) });
+        self.dsi.dltcr().modify(|_, w| unsafe {
+            w.mrd_time()
+                .bits(phy_timers.dataline_max_read_time)
+                .hs2lp_time()
+                .bits(phy_timers.dataline_hs2lp)
+                .lp2hs_time()
+                .bits(phy_timers.dataline_lp2hs)
```

Comment:
More compact and `rustfmt` friendly style to write the same
```suggestion
            w.mrd_time().bits(phy_timers.dataline_max_read_time);
            w.hs2lp_time().bits(phy_timers.dataline_hs2lp);
            w.lp2hs_time().bits(phy_timers.dataline_lp2hs)
```

Reply:
Same for other similar field chains.

Reply:
Done.  There are some other uses of chained writes in the original H7 code, left as-is.